python/tvm/relay/qnn/op/legalizations.py

```diff
@@ -35,32 +50,51 @@ def legalize_qnn_conv2d(attrs, inputs, types):
         The args of the Relay expr to be legalized
     types : list of types
         List of input and output types
+    op : Relay op
+        The original Relay op.
 
     Returns
     -------
     result : tvm.relay.Expr
         The legalized expr
+
     """
-    return qnn_conv2d_legalize(attrs, inputs, types)
+    # Collect the dtypes.
+    data_dtype = types[0].dtype
+    kernel_dtype = types[1].dtype
 
-# Generic QNN Conv2D legalization function.
-@tvm.target.generic_func
-def qnn_conv2d_legalize(attrs, inputs, types):
-    """Default legalization is None."""
-    return None
+    # Collect the input exprs.
+    data, kernel = inputs
```

Comment:
it assumes that inputs has a size 2. it would be better to do
`data, kernel = inputs[0], inputs[1]` like you did for dtype above.